### DISCUSSION TITLE
AI引擎有没有可能考虑支持一下ollama呢？

### DISCUSSION THREAD

**Discussion by BoosterCore** [ZuodaoTech/everyone-can-use-english#826] [Ideas]
AI引擎有没有可能考虑支持一下ollama呢？非常期待，感谢感谢！

**Comment by an-lee** [comment]
一开始就已经支持了的，只要本地运行了 Ollama，并且地址为 `http://localhost:11434` ，创建对话的时候就会显示 Ollama 选项

![image](https://github.com/xiaolai/everyone-can-use-english/assets/22209056/54c2bbc1-730e-43df-8493-d3e19555ed09)

**Comment by BoosterCore** [comment]
好的，多谢，我试试看

**Comment by BoosterCore** [comment]
是这样的，新建对话中确实是可以用ollama，但这个页面中并不能。目前用的OpenAI和Gemini的API都是免费版，所以在学习视频时，查看翻译，响应有些慢。
![屏幕截图 2024-05-22 163255](https://github.com/xiaolai/everyone-can-use-english/assets/73834045/e6a8ffb5-3704-4e5c-ba65-4f1f31f30c76)

**Comment by an-lee** [comment]
> 目前用的OpenAI和Gemini的API都是免费版，所以在学习视频时，查看翻译，响应有些慢。

是在用 EnjoyAI 的模型吗？响应慢？

还是说你在用自己的 OpenAI 和 Gemini 密钥？

你是想在翻译这些功能上也用 Ollama？这个目前确实还没支持。后续我再搞一下吧

**Comment by BoosterCore** [comment]
是的，我目前用的自己的密钥，但都是免费版，所以速度和token都有限制。但我本地有4090显卡，所以想在翻译时也用上ollama。多谢！

**Comment by LingJingMaster** [comment]
可以使用LM studio复用openai的接口，来达成使用纯本地ai的方式
![iShot_2024-06-25_15 40 54](https://github.com/xiaolai/everyone-can-use-english/assets/151422197/8a6c609f-1c9e-4cab-a2cf-ffbaed3a1f2c)
![iShot_2024-06-25_15 41 02](https://github.com/xiaolai/everyone-can-use-english/assets/151422197/3b892461-75e7-4ea6-bcca-037e8606140f)

**Comment by BoosterCore** [comment]
非常感谢更新，不过我本地仅部署了ollama，于是我试了`http://localhost:11434/v1/`，貌似是能检测到API，但由于默认 AI 模型列表里并不能设置相应模型，所以点翻译后报错 `404 model 'gpt-4o' not found, try pulling it first`。
不知道能否再恳请您尝试更新对ollama的支持。我这边之所以没有装 LM studio ，是因为硬盘容量实在有点紧张了，另外也不太确定 LM studio 和 ollama 这两套系统是否可以和平共处。
要求有点太高了，还望见谅！

**Reply by almensu** [reply]
![image](https://github.com/user-attachments/assets/7fa1560a-d565-419e-a682-58747426a0af)
密钥：ollama
接口地址：http://localhost:11434/v1
自定义模型如果不懂有哪些？在终端搜索命令行 ollama list 查看
